src/v/redpanda/admin/api-doc/debug.json

```diff
@@ -376,6 +402,40 @@
                 }
             }
         },
+        "memory_profile": {
+            "id": "memory_profile",
+            "description": "Sampled memory profile of a shard",
+            "properties": {
+                "shard": {
+                    "type": "long",
+                    "description": "Id of the shard the profile is from"
+                },
+                "allocation_sites": {
+                    "type": "array",
+                    "items": {
+                        "type": "allocation_site"
+                    }
+                }
+            }
+        },
+        "allocation_site": {
+            "id": "allocation_site",
+            "description": "A single allocation site with backtrace, size and count",
+            "properties": {
+                "size": {
+                    "type": "long",
+                    "description": "Current bytes allocated at this allocation site (note this is the upscaled size and not the sampled one)"
+                },
+                "count": {
+                    "type": "long",
+                    "description": "Live allocations at this site"
```

Comment:
I think think it is confusing that "size" and "count" have a totally different basis: upscaled and not. I would not expect this and I'd expect `size / count` to give the average (often "only") allocation size but it won't.

If we want to provide upscaled values (I think it is quite nice to), I suggest we provided it for both. 

`size` is also perhaps a bit unclear?

Maybe `bytes_sampled`, `bytes_upscaled`, `count_sampled`, `count_upscaled`?

Reply:
I see your point. However, I don't really see how one could upscale the count? We are not sampling based on it.

Overall this is a bit an artefact of the seastar implementation. I guess the proper names would be `bytes_upscaled` and `samples_taken`. I haven't made any use of the count value yet so we could probably just drop it all completely from the debug API.

Reply:
I think the count is really useful because the average size of the allocation is useful info.

I had assumed "count" gets scaled up in a similar way as size: by the inverse of the probability that any given allocation is sampled - but I see the problem: it's that different allocation sizes may be mixed at the same site?

Still doesn't `upscaled_size / sampled_size * sampled_count` give a meaningful number?

Reply:
>  it's that different allocation sizes may be mixed at the same site

Yeah exactly.

 > Still doesn't upscaled_size / sampled_size * sampled_count give a meaningful number?

It breaks if the allocations are very diverse, for example:

Sample Rate: 16k (lets assume we sample every 16k for simplicity)
Actual allocations: 16 * 1k (one gets sampled), 1 * 16k (gets sampled)

Math would then work out as:

Upscaled size: 32k
Sampled size: 17k
Sampled count: 2

=> Upscaled count: 32k / 17k * 2 ~ 3.7 vs. 17 actual allocations

Real avg. allocation size: 32k / 17 = 1.8k 

I will think a bit more over the weekend but I can't really come up with anything meaningful.

Reply:
You are right, I wasn't thinking this through carefully. I feel like we should track the min & max size for sampled allocations, which would help cover this hold but this can wait for another change.